docs/7/7.4-ansible.md

```diff
@@ -1,3 +1,154 @@
+  tasks:
+  - name: ensure apache is at the latest version
+    become: yes #(2)
+    yum:
+      name: httpd
+      state: latest
+```
+1. Here we are connecting as a normal user
+2. Here we tell Ansible to run the task as root. Note: you can also specify *become* as part of the *play* to set the default for all of the tasks in that *play*.
+
+You can read more about privilege escalation at [Ansible.com: Understanding Privilege Escalation](https://docs.ansible.com/ansible/latest/user_guide/become.html)
+
+#### Running Playbooks
+
+To run a playbook simply run `ansible-playbook playbook.yml` where playbook.yml is your playbook file.
+
+### Modules
+
+Modules encapsulate a single repeatable, configurable action such as controlling system resources, executing a command or making an API request. Modules can be executed directly but are usually used in playbooks where each task defines a module to use and the parameters for that module.
+
+Ansible comes with over [1300 modules](https://docs.ansible.com/ansible/latest/modules/list_of_all_modules.html) which makes it easy write tasks for just about anything. If a module does not exist for a specific task there are lower level modules for executing commands, managing system resources and making requests directly or you can write your own modules.
+
+Modules also server another purpose in that they check the state of a host to decide what, if any, action needs to be taken. This make playbooks simpler to write and also speeds them up by not repeating unnecessary actions.
+
+### Playbook variables, templates and roles
+
+You should be able to start writing basic playbooks with what you've learned so far. To write more advanced playbooks that are modular and reusable you'll need to understand some more advanced concepts.
+
+#### Variables
+
+Variables let you define values which can be used throughout your playbook to make it easier to reuse it for different hosts or environments. Variables can be defined in your inventory, playbook or can be imported from files. Read more about [variables](https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html).
+
+#### Templates
+
+Ansible uses Jinja2 to create templates which can be used by your playbook tasks or can be copied onto your hosts. Jinja2 can incorporate variable values and has a variety of filters to generate the template content. Read more about [templates](https://docs.ansible.com/ansible/latest/user_guide/playbooks_templating.html).
+
+#### Roles
+
+Roles let you organize you playbook in a more modular way by grouping task, variables, templates and other Ansible structures together. Read more about [roles](https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html).
+
+### Conclusion
+
+We have only covered a small portion of Ansible's use cases and concepts. You should also look over the [official documentation](https://docs.ansible.com/) and refer back to it working through the exercises.
+
+## Exercise 1
```

Comment:
Do we want to bullet or number the exercise?
Also do we want to use capitalize letters?